modules/ROOT/pages/dialog-components.adoc

```diff
@@ -539,6 +539,96 @@ A *textarea* is a multiline text field.
 }
 ----
 
+[[tree]]
+=== tree
+
+A *tree* is a composite component that presents an expandable category tree expandable down one level. It can also include a search and filter field.
+
+*onLeafAction:* Interacting with a *tree* component executes the `+onLeafAction+` callback set in the dialog configuration.
+
+[source,js]
+----
+tinymce.init({
+  selector: "textarea",
+  toolbar: "tree",
+  setup: (ed) => {
+    ed.ui.registry.addButton('tree', {
+      text: 'Tree toolbar button',
+      onAction: () => {
+        ed.windowManager.open({
+          title: 'Tree dialog',
+          buttons: [{type: 'cancel', text: 'Cancel',}],
+          body : {
+            type: 'panel',
+            items: [{
+              type: 'tree',
+              onLeafAction: (id) => {
+                console.log('clicked on item with id', id);
+              },
+              items: [
+                {
+                  type: 'directory',
+                  id: 'dir',
+                  title: 'Directory',
+                  menu: {
+                    icon: 'image-options',
+                    type: 'menubutton',
+                    fetch: (success) => success([
+                      {
+                        type: 'menuitem',
+                        text: 'Menu item',
+                        onAction: () => {
+                          console.log; // for demo purposes
+                        }
+                      }
+                    ])
+                  },
+                  children: [
+                    {
+                      type: 'leaf',
+                      title: 'Leaf 3',
+                      id: '3'
+                    },
+                    {
+                      type: 'leaf',
+                      title: 'Leaf 4',
+                      id: '4',
+                      menu: {
+                        icon: 'image-options',
+                        type: 'menubutton',
+                        fetch: (success) => success([
+                          {
+                            type: 'menuitem',
+                            text: 'Menu item',
+                            onAction: () => {
+                              console.log; // for demo purposes
+                            }
+                          }
+                        ])
```

Comment:
```suggestion
                        type: 'menubutton',
                        fetch: (success) => success([
			  'blocks', 'fontfamily', 'fontsize'
                        ])
```
Much simpler explanation than configuring custom menu items

Reply:
@MitchCTiny I disagree. The point of having custom menu items there is to have a custom onAction that logs something to the console that shows that it works. One can not see the effect of these predefined menu items until they close the open modal. So to test things they would have to open the modal, click on the menu item then close it again to see if what they did had any effect. This is how I see it but I may be wrong 🤷‍♂️.

Reply:
I’ve added a second example. The first presents the tree as per @hamza0867’s original.

The second shows the tree making TinyMCE menu items available in the dialog.

You could, perhaps, accuse me of attempting to split the difference, but I think there’s a specific utility here — making menu items available in a dialog — that isn’t really about presenting a tree structure as such.

Reply:
I'm happy with the use of both, including predefined menu items would show users the type of items shown here, while custom menu items show how it can be configured to their needs.